### PR TITLE
Feature/add multiple test groups

### PR DESCRIPTION
**Pull request by NasibMansour**
We haven't pulled this update to `develop` yet.
Also, I updated the documentation per issue #38 here

### CODE REVIEW

File: CommandsManual.md [hunk 1]
```diff
@@ -217,6 +219,56 @@ Attributes accepted as `<BUILD_IDENTIFIER>` (in order of search):
 
 An auth file is a `.impt.auth` file. It stores authentication and other information necessary to execute *impt* commands.
 
+**Note** An Auth file may contain more than one account's auth information, and can contain github info added using [impt auth github](#auth-github).
+
+Each account's auth information contains the following:
```
Comment (mm-gmbd) [inline CommandsManual.md]: Probably makes sense to update this section to say which keys are required/included, **based on the `profileType`**.

For example:

```md
Each account's auth information contents are dependent on the `profileType`, which is one of the keys of the auth payload. Currently, the supported types are "impCentral" or "github".

If the `profileType` is "impCentral", the following contents are included:

- `endpoint`: an impCentral API endpoint (the API base URL).
- `profileType`: "impCentral"
- `accessToken`: Account's access token.

...etc. You get the point :)
```

File: CommandsManual.md [hunk 2]
```diff
@@ -217,6 +219,56 @@ Attributes accepted as `<BUILD_IDENTIFIER>` (in order of search):
...
+- `userName`
+- `accounts`: an array of all accounts that can be accessed by this account.
+- `isDefault`: Boolean flag to indicate whether this is the default account to be used with *impt* commands. It is only included in one account and can be changed using [impt auth select](#auth-select).
+
+The Auth file is of the form:
+
+```JSON
+{
+    "USER_ACCOUNT_ID1": {
+        "endpoint": "https://api.imp.eatonem.com/v5",
```
Comment (mm-gmbd) [inline CommandsManual.md]: Considering this is public, let's remove our impCloud endpoint

File: CommandsManual.md [hunk 3]
```diff
@@ -217,6 +219,56 @@ Attributes accepted as `<BUILD_IDENTIFIER>` (in order of search):
...
+    },
+    "USER_ACCOUNT_ID2": {
+        "endpoint": "https://api.electricimp.com/v5",
+        "profileType": "impCentral",
+        "accessToken": "ACCESS_TOKEN",
+        "expiresAt": "2019-08-08T15:26:11.370Z",
+        "loginKey": "LOGIN_KEY",
+        "userName": "USERNAME",
+        "accounts": [
+            "USER_ACCOUNT_ID1",
```
Comment (mm-gmbd) [inline CommandsManual.md:268]: Is this supposed to be `USER_ACCOUNT_ID2?`

Comment (mm-gmbd) [reply CommandsManual.md:268]: And does the first account in the `accounts` list actually have to include the same ID as the key name for the user entry?

File: CommandsManual.md [hunk 4]
```diff
@@ -274,36 +326,109 @@ Each Project file contains settings for a Project, an *impt* entity which links
 
 A Project file may affect commands called from the directory where the file is located. Product, Device Group, Devices, Deployment, and source code files referenced by Project file may be assumed by a command when they are not specified explicitly.
 
+**Note** Currently, Builder information in the project file must be created manually.
+
+Each project file contains the following:
+
+- Global Builder information: includes `variables` which are used by all device groups, and `libs` which are `.js` files that contain some functions used by builder variables
+- Device Groups: Each device group is a key-value pair where the key is the device gorup ID and the value is an object containing the following information:
```
Comment (mm-gmbd) [inline CommandsManual.md]: Spell check - "gorup"

File: CommandsManual.md [hunk 5]
```diff
@@ -274,36 +326,109 @@ Each Project file contains settings for a Project, an *impt* entity which links
...
+    - `endpoint`: an impCentral API endpoint (the API base URL).
+    - `accountID`: the account ID that the device group belongs to.
+    - `deviceFile`:The device source code file name
+    - `agentFile`: The agent source code file name
+    - Local Builder information: includes `variables` which are used the specific device group, and `preBuild` and `postBuild` which are commands or `.js` files that are executed before and after building the code.
+    - `isDefault`: Boolean flag to indicate whether this is the default device group to be used . It is only included in one of device groups.
```
Comment (mm-gmbd) [inline CommandsManual.md]: There's an extra space between "used" and the period at the end of the first sentence.

Also, what happens if "isDefault" is included more than once?

The 2nd sentence should probably be slightly reworded: "It must be included in one (and only one) of the device group entries."

File: CommandsManual.md [hunk 6]
```diff
@@ -274,36 +326,109 @@ Each Project file contains settings for a Project, an *impt* entity which links
 
 A Project file may affect commands called from the directory where the file is located. Product, Device Group, Devices, Deployment, and source code files referenced by Project file may be assumed by a command when they are not specified explicitly.
 
+**Note** Currently, Builder information in the project file must be created manually.
+
+Each project file contains the following:
+
+- Global Builder information: includes `variables` which are used by all device groups, and `libs` which are `.js` files that contain some functions used by builder variables
```
Comment (mm-gmbd) [inline CommandsManual.md]: I think `preBuild` and `postBuild` can also be included in the Global builder section.

Comment (mm-gmbd) [reply CommandsManual.md]: I believe `preBuild` and `postBuild` are available in the Global builder info.

And, with that in mind, what happens if `preBuild` and `postBuild` are included in the Global and Local sections of the build file? What is the order of execution? That should be specified (and it should probably be:

Global PreBuild
Local PreBuild
Local PostBuild
Global PostBuild

Comment (mm-gmbd) [inline CommandsManual.md]: Also, I would rename "Global Builder information" to "Global Builder section"

File: CommandsManual.md [hunk 7]
```diff
@@ -274,36 +326,109 @@ Each Project file contains settings for a Project, an *impt* entity which links
...
+- Device Groups: Each device group is a key-value pair where the key is the device gorup ID and the value is an object containing the following information:
+    - `endpoint`: an impCentral API endpoint (the API base URL).
+    - `accountID`: the account ID that the device group belongs to.
+    - `deviceFile`:The device source code file name
+    - `agentFile`: The agent source code file name
+    - Local Builder information: includes `variables` which are used the specific device group, and `preBuild` and `postBuild` which are commands or `.js` files that are executed before and after building the code.
```
Comment (mm-gmbd) [inline CommandsManual.md]: Local builder information --> Local builder section

File: CommandsManual.md [hunk 8]
```diff
@@ -274,36 +326,109 @@ Each Project file contains settings for a Project, an *impt* entity which links
...
+    }
+  }
+}
+```
+
 ## Secrets Files ##
 
 A Secrets file is a `.impt.project.secrets` file located in a given directory. Different directories may contain different Secrets files. A directory can contain only one Secrets file.
 
 Each Secrets file contains builder variables for information that shouldn't be tracked in GitHub. A Secrets file may contain a global builder variables object and specific builder variables associated with the device groups.
```
Comment (mm-gmbd) [inline CommandsManual.md]: Shouldn't be tracked in ~~Github~~ git.

File: CommandsManual.md [hunk 9]
```diff
@@ -447,6 +572,22 @@ Changes default auth account.
 | --output | -z | No | Yes | Adjusts the [command's output](#command-output) |
 | --help | -h | No | No | Displays a description of the command. Ignores any other options |
 
+#### Auth Github ####
```
Comment (mm-gmbd) [inline CommandsManual.md:591]: Auth Github --> Github Auth

File: lib/Test.js [hunk 10]
```diff
@@ -165,10 +183,60 @@ class Test {
         }
     }
 
-    run(options) {
+    async run(options) {
+        var sucessAll = true;
```
Comment (mm-gmbd) [inline lib/Test.js]: typo `sucessAll` --> `successAll`

File: CommandsManual.md [hunk 11]
```diff
@@ -10,6 +10,7 @@
 **[impt auth login](#auth-login)**<br />
```
Comment (mm-gmbd) [inline CommandsManual.md:12]: You may add a general note to the top of this file, something about how some of the internals of impt have been updated to support multiple DGs, and while we have added some new features, other features have not been re-tested in full so YMMV (YMMV = your-mileage-may-vary).

Also, I'd add the same disclaimer to the README.md file so it's front and center.

**Review by mm-gmbd — CHANGES_REQUESTED** [review]

File: CommandsManual.md (hunk 3 above)
Comment (NasibMansour) [reply CommandsManual.md:268]: It's `USER_ACCOUNT_ID1`, this array has all the accounts that can be accessed by the account ID that is the key name for the user entry as returned by the command `impt account list`

File: CommandsManual.md (hunk 5 above)
Comment (NasibMansour) [reply CommandsManual.md]: if it is included more than once, then the last account in the file with `isDefault` is going to be the default.

EDIT: Nevermind, I thought this one is for the auth file, but I think should be the same here, the last device group with `isDefault` is going to be the default.

File: CommandsManual.md (hunk 6 above)
Comment (NasibMansour) [reply CommandsManual.md]: I'm sure that `preBuild` execute before `postBuild`  not matter what. However, I'm not sure as to whether the global scripts would execute before the local within `preBuild` or `postBuild` it depends on how `deepmerge` works

File: CommandsManual.md (hunk 9 above)
Comment (NasibMansour) [reply CommandsManual.md:591]: This one is correct (the command is `impt auth github`)

**Pull request merged**